SU2_CFD/src/iteration/CFluidIteration.cpp

```diff
@@ -245,6 +270,112 @@ bool CFluidIteration::Monitor(COutput* output, CIntegration**** integration, CGe
   return StopCalc;
 }
 
+void CFluidIteration::TurboMonitor(CGeometry**** geometry_container, CConfig** config_container, unsigned long ExtIter) {
+
+  su2double rot_z_ini, rot_z_final ,rot_z;
+  su2double outPres_ini, outPres_final, outPres;
+  unsigned long rampFreq, finalRamp_Iter;
+  unsigned short iMarker, KindBC, KindBCOption;
+  unsigned short iZone;
+  string Marker_Tag;
+
+  bool print;
+  if (config_container[ZONE_0]->GetMultizone_Problem())
+    ExtIter = config_container[ZONE_0]->GetOuterIter();
+
+  for(iZone = 0; iZone < nZone; iZone++) {
+  /*--- ROTATING FRAME Ramp: Compute the updated rotational velocity. ---*/
+    if (config_container[iZone]->GetGrid_Movement() && config_container[iZone]->GetRampRotatingFrame()) {
+      rampFreq       = SU2_TYPE::Int(config_container[iZone]->GetRampRotatingFrame_Coeff(1));
+      finalRamp_Iter = SU2_TYPE::Int(config_container[iZone]->GetRampRotatingFrame_Coeff(2));
+      rot_z_ini = config_container[iZone]->GetRampRotatingFrame_Coeff(0);
+      print = false;
+      if(ExtIter % rampFreq == 0 &&  ExtIter <= finalRamp_Iter){
+        rot_z_final = config_container[iZone]->GetFinalRotation_Rate_Z();
+        if(abs(rot_z_final) > 0.0){
+          rot_z = rot_z_ini + ExtIter*( rot_z_final - rot_z_ini)/finalRamp_Iter;
+          config_container[iZone]->SetRotation_Rate(2, rot_z);
+          if(rank == MASTER_NODE && print && ExtIter > 0) {
+            cout << endl << " Updated rotating frame grid velocities";
+            cout << " for zone " << iZone << "." << endl;
+          }
+          geometry_container[iZone][INST_0][MESH_0]->SetRotationalVelocity(config_container[iZone], print);
+          geometry_container[iZone][INST_0][MESH_0]->SetShroudVelocity(config_container[iZone]);
+        }
+
+        for (iZone = 0; iZone < nZone; iZone++) {
+          geometry_container[iZone][INST_0][MESH_0]->SetAvgTurboValue(config_container[iZone], iZone, INFLOW, false);
+          geometry_container[iZone][INST_0][MESH_0]->SetAvgTurboValue(config_container[iZone],iZone, OUTFLOW, false);
+          geometry_container[iZone][INST_0][MESH_0]->GatherInOutAverageValues(config_container[iZone], false);
+
+        }
+      }
+    }
+  }
+
+
+    /*--- Outlet Pressure Ramp: Compute the updated rotational velocity. ---*/
+  if (config_container[ZONE_0]->GetRampOutletPressure()) {
+    rampFreq       = SU2_TYPE::Int(config_container[ZONE_0]->GetRampOutletPressure_Coeff(1));
+    finalRamp_Iter = SU2_TYPE::Int(config_container[ZONE_0]->GetRampOutletPressure_Coeff(2));
```

Comment:
At some point you may want a more expressive way to do this, for example create a small struct for "ramps" so that you can access things by name (e.g. frequency) we have the same problem with adaptive CFL params.